src/python/nimbusml/tests/ensemble/test_votingregressor.py

```diff
@@ -0,0 +1,109 @@
+
+from nimbusml import Pipeline
+from nimbusml.ensemble import LightGbmRegressor, VotingRegressor
+from nimbusml.linear_model import (OrdinaryLeastSquaresRegressor,
+                                   OnlineGradientDescentRegressor)
+
+
+train_data = {'c1': [2, 3, 4, 5],
+              'c2': [20, 30.8, 39.2, 51]}
+train_df = pd.DataFrame(train_data).astype({'c1': np.float32,
+                                            'c2': np.float32})
+
+test_data = {'c1': [2.5, 3.5],
+             'c2': [1, 1]}
+test_df = pd.DataFrame(test_data).astype({'c1': np.float32,
+                                          'c2': np.float32})
+
+olsrArgs = {
+    'feature': ['c1'],
+    'label': 'c2',
+    'normalize': "Yes"
+}
+
+ogdArgs = {
+    'feature': ['c1'],
+    'label': 'c2',
+    'shuffle': False,
+    'number_of_iterations': 800,
+    'learning_rate': 0.1,
+    'normalize': "Yes"
+}
+
+lgbmArgs = {
+    'feature':['c1'],
+    'label':'c2',
+    'random_state':1,
+    'number_of_leaves':4,
+    'minimum_example_count_per_leaf': 1,
+    'normalize': 'Yes'
+}
+
+
+class TestVotingRegressor(unittest.TestCase):
+
+    def test_ensemble_with_average_and_median_combiner(self):
+        r1 = OrdinaryLeastSquaresRegressor(**olsrArgs)
+        r1.fit(train_df)
+        result1 = r1.predict(test_df)
+
+        r2 = OnlineGradientDescentRegressor(**ogdArgs)
+        r2.fit(train_df)
+        result2 = r2.predict(test_df)
+
+        r3 = LightGbmRegressor(**lgbmArgs)
+        r3.fit(train_df)
+        result3 = r3.predict(test_df)
+
+        r1 = OrdinaryLeastSquaresRegressor(**olsrArgs)
+        r2 = OnlineGradientDescentRegressor(**ogdArgs)
+        r3 = LightGbmRegressor(**lgbmArgs)
+
+        pipeline = Pipeline([VotingRegressor(estimators=[r1, r2, r3], combiner='Average')])
+        pipeline.fit(train_df)
+        result4 = pipeline.predict(test_df)
+
+        average1 = (result1[0] + result2[0] + result3[0]) / 3
+        average2 = (result1[1] + result2[1] + result3[1]) / 3
+        self.assertAlmostEqual(average1, result4.loc[0, 'Score'], places=5)
+        self.assertAlmostEqual(average2, result4.loc[1, 'Score'], places=5)
+
+        r1 = OrdinaryLeastSquaresRegressor(**olsrArgs)
+        r2 = OnlineGradientDescentRegressor(**ogdArgs)
+        r3 = LightGbmRegressor(**lgbmArgs)
+
+        pipeline = Pipeline([VotingRegressor(estimators=[r1, r2, r3], combiner='Median')])
+        pipeline.fit(train_df)
+        result4 = pipeline.predict(test_df)
+
+        self.assertEqual(result2.loc[0], result4.loc[0, 'Score'])
+        self.assertEqual(result1.loc[1], result4.loc[1, 'Score'])
+
+    def test_ensemble_supports_user_defined_transforms(self):
+        # TODO: fill this in
+        pass
+
+    def test_ensemble_supports_output_predictor_model(self):
+        # TODO: fill this in
+        pass
+
+    def test_ensemble_supports_cv(self):
+        # TODO: fill this in
+        pass
+
+    def test_ensemble_supports_get_fit_info(self):
+        # TODO: fill this in
```

Comment:
># TODO: fill this in [](start = 8, length = 20)

One more test where Pipeline has few transforms before VotingClassifier

Reply:
There is a unit test stub for that on line 91.

---
In reply to: [361724877](https://github.com/microsoft/NimbusML/pull/390#discussion_r361724877) [](ancestors = 361724877)